### DISCUSSION TITLE
How to keep messages after quit some git operation?

### DISCUSSION THREAD

**Discussion by gaoxinlei** [ohmyzsh/ohmyzsh#10297] [Help] [answered]
Hi, when I try to use "git log", "git diff" and other commands under "oh my zsh" environment, a new view will be opened. After I press the "q" button to exit "git log", the command execution results will be cleared, which makes me very tired. And I'm sure that won't be happen without "oh my zsh" in my mac laptop, Is there a way to keep the output of "git log" and other commands on the console while using the "oh my Zsh" plug-in.

Here is some demo:

```
commit 49bd87bf479e576c37b34a6a2290026ee489fa11 (HEAD -> master) 

Author: gaoxinlei <gaoxinlei@msn.com> 

Date:   Thu Oct 14 11:49:46 2021 +0800 

 

    demo b 

 

commit dffd0774e7782c31ffea89381486d1a413177a41 

Author: gaoxinlei <gaoxinlei@msn.com> 

Date:   Thu Oct 14 11:49:29 2021 +0800 

 

    demo a 

(END) 
```

press "q" then as here shows, there is nothing useful left:
```
➜  demo git:(master) git log 

➜  demo git:(master)   
```

You see? that's not same as the result without "oh my zsh" and not match my expect, I really want to show the command execution result for git in some cases such as when I want to compare two commits, which makes me had to open two windows both at the same time, one execute "git log" and hold it not quit, another use "git diff" and pick each commit from first window.
please tell me if there is any way to solve this, thanks a lot and wish your reply!

**Comment by mcornella** [comment, marked as answer]
I belive git is using `less` as the pager. [See this](https://superuser.com/a/106644/545959) for instructions on how to set it up to not clear on screen. 

You can also disable the git pager on certain commands. For example, [this is how you'd disable pager for `git branch`](https://stackoverflow.com/a/48370253/5798232).

**Reply by gaoxinlei** [reply]
Yeah thanks a lot, I checked this by "git help config" and saw this:
========
core.pager
           Text viewer for use by Git commands (e.g., less). The value is meant to be interpreted by the shell. The order of preference is the
           $GIT_PAGER environment variable, then core.pager configuration, then $PAGER, and then the default chosen at compile time (usually
           less).

           When the LESS environment variable is unset, Git sets it to FRX (if LESS environment variable is set, Git does not change it at all).
           If you want to selectively override Git's default setting for LESS, you can set core.pager to e.g.  less -S. This will be passed to
           the shell by Git, which will translate the final command to LESS=FRX less -S. The environment does not set the S option but the
           command line does, instructing less to truncate long lines. Similarly, setting core.pager to less -+F will deactivate the F option
           specified by the environment from the command-line, deactivating the "quit if one screen" behavior of less. One can specifically
           activate some flags for particular commands: for example, setting pager.blame to less -S enables line truncation only for git blame.

           Likewise, when the LV environment variable is unset, Git sets it to -c. You can override this setting by exporting LV with another
           value or setting core.pager to lv +c.
=============
Problem was solved  as your suggestion , thank you again.
